btrack/config.py

```diff
@@ -0,0 +1,150 @@
+    object_model : Optional[ObjectModel]
+        The `btrack` object model. See `models.ObjectModel` for more details.
+    hypothesis_model : Optional[HypothesisModel]
+        The `btrack` hypothesis model. See `models.HypothesisModel` for more
+        details.
+    max_search_radius : float
+        The maximum search radius of the algorithm in isotropic units of the
+        data. Should be greater than zero.
+    return_kalman : bool
+        Flag to request the Kalman debug info when returning tracks.
+    volume : Optional[ImagingVolume]
+        The imaging volume as [(xlo, xhi), ..., (zlo, zhi)]. See
+        `btypes.ImagingVolume` for more details.
+    update_method : constants.BayesianUpdates
+        The method to perform the bayesian updates during tracklet linking.
+            BayesianUpdates.EXACT
+                Use the exact Bayesian update method. Can be slow for systems
+                with many objects.
+            BayesianUpdates.APPROXIMATE
+                Use the approximate Bayesian update method. Useful for systems
+                with may objects.
+            BayesianUpdates.CUDA
+                Use the CUDA implementation of the Bayesian update method. Not
+                currently implemented.
+    optimizer_options: dict
+        Additional options to pass to the optimizer. See `cvxopt.glpk` for more
+        details of options that can be passed.
+
+    Notes
+    -----
+    TODO(arl): add more validation to parameters.
+    """
+
+    name: str = "Default"
+    version: str = constants.get_version()
+    verbose: bool = False
+    motion_model: Optional[MotionModel] = None
+    object_model: Optional[ObjectModel] = None
+    hypothesis_model: Optional[HypothesisModel] = None
+    max_search_radius: float = constants.MAX_SEARCH_RADIUS
+    return_kalman: bool = False
+    volume: Optional[ImagingVolume] = None
+    update_method: constants.BayesianUpdates = constants.BayesianUpdates.EXACT
+    optimizer_options: dict = constants.GLPK_OPTIONS
+
+    @validator("volume", pre=True, always=True)
+    def parse_volume(cls, v):
+        if isinstance(v, tuple):
+            return ImagingVolume(*v)
+        return v
+
+    class Config:
+        arbitrary_types_allowed = True
+        validate_assignment = True
+        json_encoders = {
+            np.ndarray: lambda x: x.ravel().tolist(),
+        }
+
+
+def load_config(filename: os.PathLike) -> TrackerConfig:
+    """Load a tracker configuration from a file.
+
+    Parameters
+    ----------
+    filename : os.PathLike
+        The filename to load the file.
+
+    Returns
+    -------
+    cfg : TrackerConfig
+        The tracker configuration.
+    """
+    logger.info(f"Loading configuration file: {filename}")
+    filename = Path(filename)
+
+    with open(filename, "r") as json_file:
+        json_data = json.load(json_file)
+
+    try:
+        cfg = _load_legacy_config(json_data)
+    except KeyError:
```

Comment:
Is there a specific key that differs? Could we check for the existence of that rather than a generic `KeyError`?

Reply:
Done!